airbyte-integrations/connectors/source-iterable/source_iterable/api.py

```diff
@@ -66,36 +78,83 @@ def request_params(self, **kwargs) -> MutableMapping[str, Any]:
 
     def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
         response_json = response.json()
-        yield from response_json.get(self.data_field, [])
+        records = response_json.get(self.data_field, [])
 
+        for record in records:
+            yield self._convert_timestamp_fields_to_datetime(record)
 
-class IterableExportStream(IterableStream, ABC):
-    def __init__(self, start_date, **kwargs):
-        super().__init__(**kwargs)
-        self._start_date = pendulum.parse(start_date)
-        self.stream_params = {"dataTypeName": self.data_field}
+    @staticmethod
+    def _parse_csv_string_to_dict(csv_string: str) -> Dict[str, Any]:
+        """
+        Parse a response with a csv type to dict object
+        Example:
+            csv_string = "a,b,c
+                          1,2,3"
 
-    cursor_field = "createdAt"
+            output = {"a": 1, "b": 2, "c": 3}
+
+
+        :param csv_string: API endpoint response with csv format
+        :return: parsed API response
+
+        """
+
+        reader = csv.DictReader(StringIO(csv_string), delimiter=",")
+        result = next(reader)
+
+        for key, value in result.items():
+            try:
+                result[key] = int(value)
+            except ValueError:
+                result[key] = float(value)
```

Comment:
Could you please explain what is it for?

Reply:
The response from the campaign_metrics endpoint is a csv file, the first line contains the name of the values, the second contains the values. There are 2 types of values: integer and floating point, so here we are casting from a string to the actual value type in a pretty naive way.